### PR TITLE
Add License tab.

### PR DESCRIPTION
**Pull request by polesye**
@auraz please review.

### CODE REVIEW

File: lms/djangoapps/lti_passport/utils.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    Returns information (`is_hidden`(bool), `children`(list)) about the xblock.
+    """
+    info = XBlockInfo(cache.get(xblock))
+    info.set_visibility(is_hidden)
+    if child is not None:
+        info.children.append(child)
+    return info
+
+
+def course_tree_info(store, simulations, access_info=None):
```
Comment (auraz) [inline lms/djangoapps/lti_passport/utils.py]: why we need it here?

**Comment by auraz** [comment]
Just curious,  can we move it to separate repo?

**Comment by auraz** [comment]
@xmig FYI means that you review if you want.

File: lms/djangoapps/lti_passport/utils.py (hunk 1 above)
Comment (polesye) [inline lms/djangoapps/lti_passport/utils.py]: To hide simulation that do not have licenses.

**Comment by polesye** [comment]
> Just curious, can we move it to separate repo?

it depends on CCX.

**Comment by polesye** [comment]
@den99f please review.

File: common/lib/xmodule/xmodule/lti_module.py [hunk 2]
```diff
@@ -526,7 +527,10 @@ def get_course(self):
         """
         Return course by course id.
         """
-        return self.descriptor.runtime.modulestore.get_course(self.course_id)
+        course_descriptor = modulestore().get_course(self.course_id)
+        course_module = self.system.get_module(course_descriptor)
+        course_module._field_data_cache = {}  # pylint: disable=protected-access
+        return course_module
```
Comment (polesye) [inline common/lib/xmodule/xmodule/lti_module.py:533]: @auraz this trick helps to fix the issue with field overrides.

**Comment by polesye** [comment]
@auraz @den99f please review.

File: cms/envs/common.py [hunk 3]
```diff
@@ -59,6 +59,7 @@
 STUDIO_NAME = "Studio"
 STUDIO_SHORT_NAME = "Studio"
 FEATURES = {
+    'CUSTOM_COURSES_EDX': True,
```
Comment (auraz) [inline cms/envs/common.py]: please move it to labster.py

Comment (auraz) [inline cms/envs/common.py]: and do not forget to add it to server vars

File: lms/djangoapps/labster_course_license/utils.py [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+            return {
+                'lti_id': match.group('lti_id'),
+                'consumer_key': match.group('consumer_key'),
+                'secret_key': match.group('secret_key'),
+            }
+        else:
+            return None
+
+    @classmethod
+    def strip_id(cls, passports):
```
Comment (auraz) [inline lms/djangoapps/labster_course_license/utils.py]: Rework this please. Using split(:)

File: lms/djangoapps/labster_course_license/views.py [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+        passports = filter(None, map(LtiPassport.parse, passports))
+        context['passports'] = passports
+        context['license'] = CourseLicense.get_license(ccx_locator)
+        context['labster_license_url'] = reverse('labster_license_handler', kwargs={'course_id': ccx_locator})
+    else:
+        context['ccx_coach_dashboard'] = reverse('ccx_coach_dashboard', kwargs={'course_id': course.id})
+    return render_to_response('labster/course_license.html', context)
+
+
+def update_descriptors(ccx, cache, descriptors):
```
Comment (auraz) [inline lms/djangoapps/labster_course_license/views.py]: please rename cache

File: lms/djangoapps/labster_course_license/views.py [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+        'user_email': user.email,
+        'license': license,
+        'source': 'edx',
+    }
+    url = settings.LABSTER_ENDPOINTS.get('consumer_secret', None)
+    response = _send_request(url, data)
+    return response.consumer_key, response.secret_key
+
+
+def get_access_info(user, course_key, lti_passports):
```
Comment (auraz) [inline lms/djangoapps/labster_course_license/views.py]: get simulation id by license

File: lms/djangoapps/labster_course_license/views.py [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+    if index is None:
+        passports.append(new_passport)
+    else:
+        passports[index] = new_passport
+
+    override_field_for_ccx(ccx, course, 'lti_passports', passports)
+    CourseLicense.set_license(ccx_locator, license)
+
+    try:
+        access_info = get_access_info(request.user, unicode(course_key), LtiPassport.strip_id(passports))
```
Comment (auraz) [inline lms/djangoapps/labster_course_license/views.py]: licensed_simulations

**Comment by denys-zarubin** [comment]
:+1:

File: lms/djangoapps/labster_course_license/utils.py (hunk 4 above)
Comment (polesye) [inline lms/djangoapps/labster_course_license/utils.py]: > Rework this please. Using split(:)

Done.

File: lms/djangoapps/labster_course_license/views.py (hunk 5 above)
Comment (polesye) [inline lms/djangoapps/labster_course_license/views.py]: > please rename cache

Done.

File: lms/djangoapps/labster_course_license/views.py (hunk 7 above)
Comment (polesye) [inline lms/djangoapps/labster_course_license/views.py]: > licensed_simulations

done

File: lms/djangoapps/labster_course_license/utils.py [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+
+
+class LtiPassportError(Exception):
+    """
+    This exception is raised in the case where Lti Passport is incorrect.
+    """
+    pass
+
+
+class LtiPassport(object):
```
Comment (polesye) [inline lms/djangoapps/labster_course_license/utils.py:21]: @auraz is it better?

**Pull request merged**